db/types/base.py

```diff
@@ -3,18 +3,60 @@
 from sqlalchemy import create_engine
 
 from db import constants
-
 from db.functions import hints
 
 from frozendict import frozendict
 
+from collections.abc import Sequence, Mapping
+
+from db.types import categories
+
+
+class DatabaseType:
+
+    value: str
+
+    @property
+    def id(self) -> str:
+        """
+        Here we're defining Enum's value attribute to be the database type id.
```

Comment:
Name might similarly collide with a type's display name. I like using `id`, because it denotes that its function is identification. If passing around `name`, it might be ambiguous whether or not that it is a suitable identifier (i.e. whether it's unique and constant).